### PR TITLE
READMEs

### PR DESCRIPTION
**Pull request by NelsonVides**
Progress is explained in each commit. First thing was to make the `mongooseim.toml` template configurable from the helm command. Then I moved info from the main README to each package's README, converted some k8s info to its equivalent helm version, updated the package versions according to the amount of changes (MIM deserves a +0.1 as the flag to add the config file is new, MPush only gets a +0.0.1 as the only change is basically the readme).

### CODE REVIEW

File: MongooseIM/README.md [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+To uninstall, simply run `helm uninstall my-mongooseim`, where `my-mongoose` is the release name you've given to your installation.
+
+## Common parameters
+
+| Parameter          | Description                                          | Default      |
+|--------------------|------------------------------------------------------|--------------|
+| `nodeName`         | Name of the nodes as containers                      | `mongooseim` |
+| `nodeCookie`       | Cookie to be used by the BEAM                        | `mongooseim` |
+| `replicaCount`     | Default number of replicas to be clustered           | `1`          |
+| `loadBalancerIP`   | Exposed external IP address for the Load Balancer    | not set (will be automatically assigned) |
```
Comment (DenysGonchar) [inline MongooseIM/README.md:36]: that would be nice to mention that it's XMPP TCP interface

File: README.md [hunk 2]
```diff
@@ -1,331 +1,60 @@
...
-watch kubectl get node,pod,pv,pvc,sts,svc
-```
-
-This is going to be our monitoring window. We'll input commands in another terminal.
+MongooseIM pods use the already existing MongooseIM container available from [MongooseIM's DockerHub][MIM-docker] or [GitHub][MIM]. Likewise, MongoosePush uses the containers available from [Mongoose Push' dockerhub][MPush-docker].
 
-If you're reading this in your browser, it's time to clone the repo:
+  * [Add this repo to your local helm](#add-this-repo-to-your-local-helm)
+  * [Initialise a K8S cluster](#initialise-a-k8s-cluster)
+  * [How to recover the plain k8s files](#how-to-recover-the-plain-k8s-files)
```
Comment (DenysGonchar) [inline README.md:9]: why `recover`? we haven't broken them :) i would say `how to get the plain...`

File: README.md [hunk 3]
```diff
@@ -1,331 +1,60 @@
...
-
-After a while it should appear in our monitoring window:
-
-```
-NAME                        TYPE        CLUSTER-IP     EXTERNAL-IP   PORT(S)                                        AGE
-service/kubernetes          ClusterIP   10.43.240.1    <none>        443/TCP                                        3h
-service/mongoose            ClusterIP   None           <none>        4369/TCP,5222/TCP,5269/TCP,5280/TCP,9100/TCP   1h
-service/mongoose-nodeport   NodePort    10.43.253.16   <none>        5222:32709/TCP                                 41s
+kubectl config use-context CONTEXT-NAME
+kubectl config use-context docker-desktop  # if Docker Desktop with Kubernetes is enabled
```
Comment (DenysGonchar) [inline README.md:44]: `if Docker Desktop with Kubernetes is enabled` -  it sounds weird, can we rephrase it?

**Review by DenysGonchar — CHANGES_REQUESTED** [review]
Please add links to the MIM and MongoosePush charts' READMEs to the main README file of the repo

File: README.md (hunk 3 above)
Comment (DenysGonchar) [reply README.md:44]: overall, all the READMEs look nice and informative. thanks a lot for the changes.

**Comment by DenysGonchar** [comment]
overall, all the READMEs look nice and informative. thanks a lot for the changes.

**Review by DenysGonchar — APPROVED** [review]

File: MongooseIM/README.md [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+
+This chart will install a MongooseIM cluster composed of a single node (set `replicaCount=n` for more nodes), sitting behind a regular k8s load-balancer, with a default sane configuration.
+
+To uninstall, simply run `helm uninstall my-mongooseim`, where `my-mongoose` is the release name you've given to your installation.
+
+## Common parameters
+
+| Parameter          | Description                                          | Default      |
+|--------------------|------------------------------------------------------|--------------|
+| `nodeName`         | Name of the nodes as containers                      | `mongooseim` |
```
Comment (chrzaszcz) [inline MongooseIM/README.md:33]: This has no effect - `start.sh` just overwrites it.

File: MongooseIM/README.md [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+| `nodeCookie`       | Cookie to be used by the BEAM                        | `mongooseim` |
+| `replicaCount`     | Default number of replicas to be clustered           | `1`          |
+| `loadBalancerIP`   | Exposed external IP address for the Load Balancer, it exposes the XMPP TCP interface | not set (will be automatically assigned) |
+| `vmConfig`         | User-given `vm.args` file                            | not set (internal default, check sources) |
+| `mimConfig`        | User-given `mongooseim.toml` configuration file      | not set (internal default, check sources) |
```
Comment (chrzaszcz) [inline MongooseIM/README.md:38]: I would somehow underline that this is the main MongooseIM config file while `vm.args` is for tweaking the Erlang VM itself, so usually only `mongooseim.toml` needs to be provided.

File: MongooseIM/README.md [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+pod/mongooseim-2   1/1       Running   0          1m
+
+NAME                          DESIRED   CURRENT   AGE
+statefulset.apps/mongooseim   3         3         4m
+```
+
+Once all the pods are up we can check the cluster status:
+
+```sh
+$ kubectl exec mongoose-0 /usr/lib/mongooseim/bin/mongooseimctl mnesia running_db_nodes
```
Comment (chrzaszcz) [inline MongooseIM/README.md:82]: There's no pod `mongoose-0` here, it's `mongooseim-0`, btw I could not get them to cluster, I am not sure why.

File: MongooseIM/README.md [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+```
+
+### Direct access via external worker node IPs
+
+The LoadBalancer is the more convenient, but also the more expensive way of publishing a service. We can also access our MongooseIM instances by using Kubernetes worker nodes' external IP addresses.
+
+In order to do that, we first have to create a NodePort service:
+
+```sh
+helm upgrade mim MongooseIM --set nodeport.enabled=true
```
Comment (chrzaszcz) [inline MongooseIM/README.md:138]: Previous `upgrade` exaple used `mongoose/mongooseim` but this one uses `MongooseIM` - is this on purpose? I think the user might get a bit lost and not understand that we moved to the cloned-repo version here.

**Review by chrzaszcz — COMMENTED** [review]
I tried to follow the steps and encountered a few issues, could you take a look?

File: MongooseIM/README.md (hunk 6 above)
Comment (NelsonVides) [reply MongooseIM/README.md]: Name fixed, here and in other places I missed. Like, the very `mongoose-svc.yaml` that defines the clustering logic, a very long time ago I forgot to update the name of the app there!

**Review by chrzaszcz — APPROVED** [review]
All systems go now, checked. Great work!

**Pull request merged**